miden-tx/src/error.rs

```diff
@@ -166,151 +169,193 @@ impl fmt::Display for AuthenticationError {
+const ERR_ACCOUNT_NONCE_INCREASE_MUST_BE_U32_VALUE: u32 = 131131;                                   // 131131 0x0002003B
+const ERR_ACCOUNT_POW_IS_INSUFFICIENT: u32 = 131135;                                                // 131135 0x0002003F
+const ERR_ACCOUNT_SEED_DIGEST_MISMATCH: u32 = 131134;                                               // 131134 0x0002003E
+
+// --- NONCE ERROR --------------------------------------------------------------------------------
+const ERR_NONCE_DID_NOT_INCREASE_AFTER_STATE_CHANGED: u32 = 131081;                                 // 131081 0x00020009
+
+// --- VAULT ERRORS -------------------------------------------------------------------------------
+const ERR_VAULT_ADDING_FUNGIBLE_ASSET_WOULD_EXCEED_MAX_AMOUNT_OF_9223372036854775807: u32 = 131117; // 131117 0x0002002D
+const ERR_VAULT_ADD_FUNGIBLE_ASSET_TO_ACCOUNT_FAILED_INITIAL_VALUE_INVALID: u32 = 131118;           // 131118 0x0002002E
+const ERR_VAULT_GET_BALANCE_PROC_CAN_BE_CALLED_ONLY_WITH_FUNGIBLE_FAUCET: u32 = 131115;             // 131115 0x0002002B
+const ERR_VAULT_HAS_NON_FUNGIBLE_ASSET_PROC_CAN_BE_CALLED_ONLY_WITH_NON_FUNGIBLE_ASSET: u32 =       // 131116 0x0002002C
+    131116;
+const ERR_VAULT_NON_FUNGIBLE_ASSET_ALREADY_EXISTS: u32 = 131119;                                    // 131119 0x0002002F
+const ERR_VAULT_REMOVE_FUNGIBLE_ASSET_TO_ACCOUNT_FAILED_INITIAL_VALUE_INVALID: u32 = 131121;        // 131121 0x00020031
+const ERR_VAULT_REMOVING_FUNGIBLE_ASSET_RESULTS_IN_UNDERFLOW: u32 = 131120;                         // 131120 0x00020030
+const ERR_VAULT_REMOVING_INEXISTENT_NON_FUNGIBLE_ASSET: u32 = 131122;                               // 131122 0x00020032
+
+// --- FAUCET ERRORS ------------------------------------------------------------------------------
+const ERR_FAUCET_BURN_CALLED_ON_NONEXISTENT_TOKEN: u32 = 131110;                                    // 131110 0x00020026
+const ERR_FAUCET_BURN_CANNOT_EXCEED_EXISTING_SUPPLY: u32 = 131107;                                  // 131107 0x00020023
+const ERR_FAUCET_MINT_WOULD_CAUSE_ISSUANCE_OVERFLOW: u32 = 131106;                                  // 131106 0x00020022
+const ERR_FAUCET_NON_FUNGIBLE_BURN_CALLED_ON_WRONG_FAUCET_TYPE: u32 = 131109;                       // 131109 0x00020025
+const ERR_FAUCET_NON_FUNGIBLE_TOKEN_ALREADY_EXISTS: u32 = 131108;                                   // 131108 0x00020024
+const ERR_FAUCET_STORAGE_DATA_SLOT_254_IS_RESERVED: u32 = 131072;                                   // 131072 0x00020000
+
+// --- ASSET GENERAL ERROR ------------------------------------------------------------------------
+const ERR_ASSET_EXCEED_MAX_AMOUNT_OF_9223372036854775807: u32 = 131138;                             // 131138 0x00020042
+
+// --- FUNGIBLE ASSET ERRORS ----------------------------------------------------------------------
+const ERR_FUNGIBLE_ASSET_DISTRIBUTE_WOULD_CAUSE_MAX_SUPPLY_TO_BE_EXCEEDED: u32 = 131105;            // 131105 0x00020021
```

Comment:
I think this also might not really be a part of the kernel.

Reply:
I'm not sure where to place this error (and also the `ERR_ASSET_EXCEED_MAX_AMOUNT_OF_9223372036854775807` above it). They could be described as "faucet" and "asset" errors respectively, but creating a new section with a new prefix just for one error looks like an overkill for me.